### DISCUSSION TITLE
Type information 'leaking' between models

### DISCUSSION THREAD

**Discussion by jjhuff** [microsoft/monaco-editor#3665] [Q&A]
When calling `setModel`, it seems like typescript info isn't being cleared properly.
Originally filed here: https://github.com/suren-atoyan/monaco-react/issues/308#

**monaco-editor version:** 0.30.1 (playground version)
**Browser:** Chrome M96
**OS:** Linux
**Playground code that reproduces the issue:**

```js
const model1 = monaco.editor.createModel(`
const foo=1;
`, 'javascript', 'root/js/file1.js');

const model2 = monaco.editor.createModel(`
const foo=2;
const bar=3;
`, 'javascript', 'root/js/file2.js');

const editor = monaco.editor.create(document.getElementById('container'), {
    model: model1,
});

setTimeout(() => {
  editor.setModel(model2);
}, 500);

setTimeout(() => {
  editor.setModel(model1);
}, 1000);
```

After the snipped toggles the models, try typing `const bar`. You'll get `bar` as a completion choice even though `bar` is defined in the non-current model.

**Comment by alexdima** [comment]
This is as designed. There is a single language service (a conceptual equivalent of a single `tsconfig.json`) instantiated per page.

**Comment by jjhuff** [comment]
Gotcha. I've been playing with this scenario in VSCode. It seems like the language service behavior matches the compiler when not using modules (I just added `export {}` to my files). They assume global scope.

However, when using modules, the errors regarding 'foo' go away. However, `bar` is still suggested as a completion in file1, even though it's not defined in that file.
